app/queues/certificate/order-creator-worker.server.ts

```diff
@@ -14,28 +15,34 @@ export interface OrderCreatorData {
 
 export interface OrderCreatorOutputData {
   certificateId: number;
-  username: string;
-  rootDomain: string;
-  orderUrl: string;
-  challengeEntries: {
-    id: number;
-    domain: string;
-    challengeKey: string;
-  }[];
 }
 
 export const orderCreatorQueueName = 'certificate-createOrder';
 
 /**
  * This async fn handles adding challenges to DB and DNS
  */
-const handleChallenges = (certificateId: number, bundles: ChallengeBundle[]) => {
+const handleChallenges = ({
+  username,
+  certificateId,
+  bundles,
+}: {
+  username: string;
+  certificateId: number;
+  bundles: ChallengeBundle[];
+}) => {
   const challengeInsertPromises = bundles.map(async ({ domain, value: challengeKey }) => {
     logger.info(`Adding challenge to DNS`, { domain, challengeKey });
 
     /**
-     * TODO actually add challenge to DNS
+     * add challenge to DNS
      */
+    await addDnsRequest({
+      username,
```

Comment:
@dadolhay  I'm going to break down this function into 1. Add a record to DB with pending status 2. Route53 update and update DB status once Route53 work is complete. Let me know if you see any concern with this.

Reply:
@Genne23v 
It's still going to be a flow, right? so I create a job to handle this, and the job flow adds the db entry, handles the route53 api, then updates the status to complete

Am I understanding you correctly?

Reply:
Initial DB record creation won't be a part of flow as it completes the work immediately. Then the rest of jobs will be run as flow.